doc/develop/sca/codechecker.rst

```diff
@@ -0,0 +1,71 @@
+.. _codechecker:
+
+CodeChecker support
+###################
+
+`CodeChecker <https://codechecker.readthedocs.io/>`__ is a static analysis infrastructure.
+It executes analysis tools available on the build system, such as
+`Clang-Tidy <http://clang.llvm.org/extra/clang-tidy/>`__,
+`Clang Static Analyzer <http://clang-analyzer.llvm.org/>`__ and
+`Cppcheck <https://cppcheck.sourceforge.io/>`__. Refer to the analyzer's websites for installation
```

Comment:
During my testing, `cppcheck` was included in the results..

Reply:
The `CodeChecker log` command would catch all compiler steps and generate a `compile_commands.json`, however this can  be simply done with `cmake` with `set(CMAKE_EXPORT_COMPILE_COMMANDS ON)`

Then `CodeChecker analyze` will run all supported and installed checkers with their respective defaults. If a users wants a different configuration, the `_OPTS` can be used for that.